core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -715,7 +717,10 @@ class LogTest extends JUnitSuite {
     // files should be renamed
     val segments = log.logSegments.toArray
     val oldFiles = segments.map(_.log.file) ++ segments.map(_.index.file)
-    log.deleteOldSegments((s) => true)
+    // expire all segments
+    log.logSegments.foreach(_.lastModified = time.milliseconds - 1000L)
```

Comment:
Yes as the logic in `Log` is, and always has been:
```
val startMs = time.milliseconds
deleted += deleteOldSegments(startMs - _.lastModified > config.retentionMs)
```
So, depending on how quick the test runs etc, `startMs - _.lastModified == 0`